crates/red_knot_python_semantic/src/unpack.rs

```diff
@@ -76,6 +76,11 @@ impl<'db> UnpackValue<'db> {
         matches!(self, UnpackValue::Iterable(_))
     }
 
+    /// Returns `true` if the value is being assigned to a target.
+    pub(crate) const fn is_assign(self) -> bool {
```

Comment:
I wanted to keep this consistent with `is_iterable` so I created this. I think https://docs.rs/is-macro/latest/is_macro/ also works here.